packages/matchbox/src/components/TextField/TextField.js

```diff
@@ -1,179 +1,207 @@
+
+  React.useLayoutEffect(() => {
+    if (prefixRef.current) {
+      setPrefixPadding(roundToBaseline(getRectFor(prefixRef.current).width + 16));
+    }
+    if (suffixRef.current) {
+      setSuffixPadding(roundToBaseline(getRectFor(suffixRef.current).width + 16));
+    }
+  }, [prefix, suffix, prefixRef, suffixRef]);
+
+  const inputProps = {
+    ...componentProps,
+    as: multiline ? 'textarea' : 'input',
+    height: multiline ? 'auto' : '2.5rem',
+    name,
+    id,
+    type,
+    disabled,
+    readOnly,
+    autoFocus,
+    placeholder,
+    onFocus,
+    onBlur,
+    onChange,
+    pl: prefixPadding,
+    pr: suffixPadding,
+    textAlign: align,
+    ...describedBy,
+    style: {
+      ...(multiline ? { resize } : {}),
+      ...style,
+    },
+  };
 
-    const labelMarkup = (
-      <Label id={id} label={`${label}${requiredIndicator}`}>
-        {error && errorInLabel && <Error className={styles.InlineError} wrapper='span' error={error} />}
+  return (
+    <StyledWrapper {...systemProps}>
+      <Label id={id} label={label} labelHidden={labelHidden}>
+        {required && (
+          <Box as="span" pr="200" aria-hidden="true">
+            *
+          </Box>
+        )}
+        {error && errorInLabel && (
+          <Box as={Error} id={errorId} wrapper="span" error={error} fontWeight="400" />
+        )}
       </Label>
-    );
-
-    const helpMarkup = helpText
-      ? <div className={styles.HelpText}>{helpText}</div>
-      : null;
-
-    const prefixMarkup = prefix
-      ? <span className={prefixClasses} ref={(node) => this.prefixNode = node}>{prefix}</span>
-      : null;
-
-    const suffixMarkup = suffix
-      ? <span className={suffixClasses} ref={(node) => this.suffixNode = node}>{suffix}</span>
-      : null;
-
-    const input = React.createElement(multiline ? 'textarea' : 'input', {
-      ...rest,
-      name,
-      id,
-      type,
-      disabled,
-      readOnly,
-      autoFocus,
-      placeholder,
-      onFocus,
-      onBlur,
-      onChange,
-      className: setInputClasses,
-      style: {
-        textAlign: align,
-        paddingLeft,
-        paddingRight,
-        resize,
-        ...style
-      }
-    });
-
-    const inputMarkup = prefix || suffix
-      ? <div className={styles.InputWrapper}>
-        {prefixMarkup}
-        {input}
-        {suffixMarkup}
-      </div>
-      : input;
-
-    return (
-      <fieldset className={styles.TextField}>
-        {label && !labelHidden && labelMarkup}
-        <Connect left={connectLeft} right={connectRight}>
-          {inputMarkup}
-        </Connect>
-        {error && !errorInLabel && <Error error={error} />}
-        {helpMarkup}
-      </fieldset>
-    );
-  }
+      <Connect left={connectLeft} right={connectRight}>
+        <Box position="relative">
+          <PrefixOrSuffix
+            content={prefix}
+            className={prefixClassname}
+            forwardedRef={prefixRef}
+            left="300"
```

Comment:
When focusing a field with a prefix icon, the icon disappears. Is that expected?

<img width="338" alt="Screen Shot 2020-03-11 at 12 26 28 PM" src="https://user-images.githubusercontent.com/3613392/76439962-90dd8380-6393-11ea-965e-61fb5e49d2e1.png">

Reply:
Fixed when using `&:focus-within` instead of `*:focus`

Reply:
<img width="299" alt="Screen Shot 2020-03-13 at 11 36 33 AM" src="https://user-images.githubusercontent.com/3613392/76636148-f1e59280-651e-11ea-9a5c-216bbdf6069c.png">

Hmm, now the text is overlapping. Probably because of those padding adjustments you made?